public/search.js

```diff
@@ -0,0 +1,204 @@
+  * @param end: the index of the last document you want to display
+  */
+  function displayDocs( start, end)
+  {
+    $("#bulletList").empty();
+    var e = document.getElementById("dropdown");
+    var selectedColl = e.options[e.selectedIndex].value;
+
+    //call to get all documents (excluding .lsqt) from the collection selected in the drop down list
+    var docs = $.ajax(
+    {
+      url: "http://localhost:3000/v1/search?structuredQuery={%20%22search%22:{%20%22query%22:{%20%22and-not-query%22:%20{%20%22positive-query%22:%20{%20%22collection-query%22:%20{%20%22uri%22:%20[%20%22"+selectedColl+"%22%20]%20}%20},%20%22negative-query%22:%20{%20%22collection-query%22:%20{%20%22uri%22:%20[%20%22lsqt%22%20]%20}%20}%20}%20},%20%22options%22:{%20%22search-option%22:[%22unfiltered%22]%20}%20}%20}&format=json&pageLength=1000",
+        success: function(data, textStatus) {
+        console.log('got collections: ' + data);
+      },
+      error: function(jqXHR, textStatus, errorThrown) {
+        console.log('problem');
+      },
+      async: false,
+    });
+    docs = JSON.parse( docs.responseText );
+
+    //Checks and sets boundary points.
+    //Looks at the index of the first and last document (passed into the function)
+    //and disables or enables the next/previous buttons based on those indexes.
+    if( start <= 1 )
+    {
+      document.getElementById("prev").disabled = true;
+    }
+    else
+    {
+      document.getElementById("prev").disabled = false;
+    }
+
+    if( end >= docs.total)
+    {
+      document.getElementById("next").disabled = true;
+      end = docs.total;
+    }
+    else
+    {
+      document.getElementById("next").disabled = false;
+    }
+
+    var docLen = docs.total;
+    document.getElementById("numDocs").innerHTML = start.toString() + " to " + end.toString() + " of " + docLen;
+
+    //adds all document URI's into an array, in order to display the results alphabetically sorted
+    var uriArray = [];
+    for (var k = 0; k < docs.total; k++) {
+      uriArray[k] = docs['results'][k]['uri'];
+    }
+    uriArray.sort();
+
+    //nested for loop to loop through the different URI's in the sorted array of URI's, and add the URI to the bullet list.
+    //then under each uri the system start, system end, valid start, valid end times will be displayed
+    //NOTE: this outer loop will not display every URI in the array, it only displays a specified range 
+    for (var i=start; i < end; i++)
+    {
+       var uri = uriArray[i];
+       //getting a list of collections the document belongs to for each URI
+       var docColl = $.ajax(
+        {
+          url: "http://localhost:3000/v1/documents?uri="+uri+"&category=collections&format=json",
+            success: function(data, textStatus) {
+            console.log('got collections: ' + data);
+          },
+          error: function(jqXHR, textStatus, errorThrown) {
+            console.log('problem');
+          },
+          async: false,
```

Comment:
This is what I did instead, would this be okay? 
      var collArr;
      var collArrLen;
      var docColl = $.ajax(
      {
        url: "/v1/documents?uri="+uri+"&category=collections&format=json",
          success: function(data, textStatus) {
          console.log('got collections: ' + data );
          collArr = data.collections;
          collArrLen = collArr.length;
        },
        error: function(jqXHR, textStatus, errorThrown) {
          console.log('problem');
        },